### PR TITLE
feat(discover) Add display modes to discover chart

### PR DESCRIPTION
**Pull request by markstory**
We'd like to expand the number of visualization options we offer. Those new options will be incompatible with previous + release markers. Moving the current options into a dropdown will also enable us to reduce the noise and clutter that release markers and previous period data can introduce.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/events/eventsChart.jsx [hunk 1]
```diff
@@ -162,6 +164,52 @@ class EventsChart extends React.Component {
...
+            currentSeriesName={currentSeriesName}
+            previousSeriesName={previousSeriesName}
+          />
+        </TransitionChart>
+      );
+    };
+
+    if (!disableReleases) {
+      const previousChart = chartImplementation;
+      chartImplementation = chartProps => (
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/events/eventsChart.jsx:206]: I don't love this pattern but couldn't come up with something better that wasn't also more complicated/indirect.

File: src/sentry/static/sentry/app/views/eventsV2/chartFooter.tsx [hunk 2]
```diff
@@ -15,21 +15,27 @@ type Props = {
...
+  onDisplayChange,
 }: Props) {
   const elements: React.ReactNode[] = [];
 
   elements.push(<SectionHeading key="total-label">{t('Total Events')}</SectionHeading>);
   elements.push(
     total === null ? (
       <SectionValue data-test-id="loading-placeholder" key="total-value">
-        -
+        &mdash;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/chartFooter.tsx:38]: Normalized these loading indicators on an mdash.

**Comment by wmak** [comment]
I think this reintroduces some loading jank when you flip between graphs, looks fine still when changing the period though :+1::
loading jank

**Comment by markstory** [comment]
@wmak Yeah, I'm not sure how to get rid of that reload jank just yet.

**Review by wmak — APPROVED** [review]
LGTM

**Pull request merged**